CRM/Upgrade/Incremental/php/FiveThirtySeven.php

```diff
@@ -106,4 +106,33 @@ public static function alterSavedSearchFK(CRM_Queue_TaskContext $ctx) {
     return TRUE;
   }
 
+  /**
+   * Update DB defaults for contribution recur.
+   *
+   * This adds default values for start_date, create_date, modified_date
+   * and frequency_interval in line with what is in the UI (frequency_unit
+   * already has 'month' as the default.
+   *
+   * The default of 'Pending' for contribution_recur_id will be updated as
+   * appropriate as soon as a contribution is attached to it by BAO code.
+   *
+   * The core code does not rely on the defaults for any of these fields.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function updateDBDefaultsForContributionRecur(CRM_Queue_TaskContext $ctx): bool {
+    $pendingID = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_ContributionRecur', 'contribution_status_id', 'Pending');
+    CRM_Core_DAO::executeQuery("
+      ALTER TABLE `civicrm_contribution_recur`
+      MODIFY COLUMN `start_date` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'The date the first scheduled recurring contribution occurs.',
```

Comment:
Should these date columns be datetime or timestamps?

Reply:
@seamuslee001 I tested & the default works with datetime - which is what they already are. I wanted to leave any type change out of scope

Reply:
I too prefer to have separate PR/issue for changing default value compared to changing the type.

Reply:
@JoeMurray I did have the PR for that change (linked above) - I closed it to incorporate into this as I felt it had been canvassed & upgrade scripts don't play well as separate PRs